test/Concurrency/isolated_any.swift

```diff
@@ -83,3 +83,16 @@ func extractFunctionIsolation(_ fn: @isolated(any) @escaping () async -> Void) {
   let _: (any Actor)? = extractIsolation(myActor.asyncThrowsActorFunction)
   let _: (any Actor)? = extractIsolation(myActor.actorFunctionWithArgs(value:))
 }
+
+func extractFunctionIsolationExpr(
+  _ fn1: @isolated(any) @escaping () async -> Void,
+  _ fn2: @isolated(any) @escaping (Int, String) -> Bool
```

Comment:
Could you please add a few test-cases that deal with optional unwrap (both styles) as well?